lib/vanagon/engine/scheduled.rb

```diff
@@ -0,0 +1,27 @@
+require 'vanagon/engine/base'
+require 'json'
+
+class Vanagon
+  class Engine
+    # Class to use when building on a shared resource device that is managed
+    # by an external scheduler.
+    class Scheduled < Base
```

Comment:
I'd be generally :+1: on using `Scheduled` for both `pooler` and `hardware`.  Lemme :zzz: on it and see what the ramifications are. I think it's probably pretty clean, though.